generate-stats

```diff
@@ -100,6 +102,34 @@ def main():
     # Sort tables
     all_tables = sorted(all_tables, key=lambda x: x["data"][0][1], reverse=True)
 
+    # Select unexpected messages
+    messages = cursor.execute("""\
+            SELECT retry_reason, TestRuns.url
+            FROM Failures
+            JOIN TestRuns ON Failures.run = TestRuns.id
+            WHERE TestRuns.time > ? AND
+                retry_reason IS NOT null AND
+                retry_reason IS NOT 'be robust against unstable tests' AND
+                retry_reason IS NOT 'due to failure of test harness or framework';""", (since,)).fetchall()
+
+    messages = [m for m in messages if m[0] is not None]
+    # Replace any numbers > 100 with 0 (usually time stamps or PIDs)
+    messages = [(re.sub(r'\d\d[\d\.]+', '0', m[0]), m[1]) for m in messages]
+
+    # Replace any partial paths like dracut.xf8hG with dracut.xxx
+    messages = [(re.sub(r'\/([^\/]*)\.[^\/]*\/', r'/\1.xxx/', m[0]), m[1]) for m in messages]
```

Comment:
No need to escape the slashes. However, this is way too vague -- it matches on non-contiguous paths/multiple words as well, such as `/ (has failed) - Exiting.`. Does it hurt much to drop this for now?

If it creates too much noise, please refine the RE to only match `/[^/ \t]{7,}\.`, i.e. non-trivial absolute paths without spaces.

Reply:
omitting it all together creates way too much noise and messages are not group almost at all.

Where does the `7` come from? With this pattern nothing again is matched, as the worst ofender is `dracut.asdfgh`, where neither part is 7 characters long.

I can change it to `/[^/ \t]{3,}\.)[^/ \t]+` which not sure if is better readable really. I want to show it is part of path, thus starts with /, then contains something, then dot and then there is some another part, that we want to replace.

Reply:
Right, but `[^/]*` is a terribly wide pattern which can basically be *anything*. Hence let's at least require that it's a single word.

The 7 was fairly arbitrary, to make sure we are not accidentally screwing up very short words, or things which are not paths at all.

Reply:
Updated

Reply:
This still looks a bit sloppy, as it replaces "static/readable" names like /foo.bar wit .xxx as well. But again, this is fungible at any time and not an API, so it's fine. Thanks!